### PR TITLE
OM-721 | Multiple phones

### PR DESCRIPTION
**Pull request by tuovinensanttu**
- Added multiple phone support: user can add, view, edit, remove and change primary phones. 
- Made changes to `updateMutationVariables` helper function. It's now as generic as possible.
- Updated / added tests.

### CODE REVIEW

File: src/profile/helpers/__tests__/updateMutationVariables.test.ts [hunk 1]
```diff
@@ -66,113 +69,176 @@ const addresses: Address[] = [
...
-  });
-
-  test('removeEmails exists', () => {
-    const emailObj = getEmail([], myProfile);
-    expect(emailObj.removeEmails).toEqual(['123', '234']);
-  });
+const phones: Phone[] = [
+  {
+    id: '123',
+    phone: '0501234567',
```
Comment (badrange) [inline src/profile/helpers/__tests__/updateMutationVariables.test.ts:75]: Do we have any tests with foreign phone numbers or addresses?

File: src/profile/helpers/updateMutationVariables.ts [hunk 2]
```diff
@@ -28,139 +36,146 @@ type AddressInputs = {
...
-      );
+const getPrimaryValue = (primary: Primary, profile?: MyProfileQuery) => {
+  const primaryValue = profile?.myProfile && profile.myProfile[primary];
+  return primaryValue || { id: '' };
+};
 
-      return address.id && !isEqual(address, profileAddress);
-    })
-    .map(address => {
+const getEmptyObject = (primary: Primary) => {
```
Comment (badrange) [inline src/profile/helpers/updateMutationVariables.ts:53]: It's not obvious what this is for, maybe a comment explaining the purpose could be helpful!

**Review by badrange — APPROVED** [review]
Looks solid! 

Could be a good idea to add a foreign phone number for the tests and some code comments.

File: src/profile/helpers/__tests__/updateMutationVariables.test.ts (hunk 1 above)
Comment (tuovinensanttu) [reply src/profile/helpers/__tests__/updateMutationVariables.test.ts:75]: There isn't currently any validation for phone numbers and the only address based validation is that postalCode is at the most 5 characters. So at the moment every phone number will pass. But if we end up adding stricter validation, testing foreign values is a good idea!

File: src/profile/helpers/updateMutationVariables.ts (hunk 2 above)
Comment (tuovinensanttu) [reply src/profile/helpers/updateMutationVariables.ts:53]: I'll add a comment. My imagination was running on fumes when I was naming these 😄

**Review by Jokinen — APPROVED** [review]
Bernt beat me to it! Looking good as far as I am concerned.

One thing I am thinking about is the UX in the view for additional fields. The controls are at the very bottom, but the new fields are always generated in a grouped order where emails are first, then addresses and so on. So if a user has an additional address and they add an email, it may be enough distance away from the button that it's not obvious.

I imagine Anniina will give more details on these at some point so I don't think it's something we should address now. Just barfing my thoughts :D

**Pull request merged**